### PR TITLE
[CDAP-17334] Add Unit Tests for LogViewer DataFetchers

### PR DESCRIPTION
**Pull request by elfenheart**
JIRA: https://issues.cask.co/browse/CDAP-17334
Build: https://builds.cask.co/browse/CDAP-UDUT856

### CODE REVIEW

File: cdap-ui/app/cdap/components/LogViewer/DataFetcher/__tests__/PreviewDataFetcher.test.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+      dataFetcher.getLast().subscribe();
+      expect(prevLogsSpy).toHaveBeenCalledWith(BASE_LOGS_PARAMS);
+    });
+    prevLogsSpy.mockRestore();
+  });
+
+  it('getFirst should fetch logs without offset', () => {
+    const nextLogsSpy = jest.spyOn(MyPreviewApi, 'nextLogs');
+    dataFetcher.init().subscribe(() => {
+      dataFetcher.getLast().subscribe();
```
Comment (njbriggs) [inline cdap-ui/app/cdap/components/LogViewer/DataFetcher/__tests__/PreviewDataFetcher.test.ts]: Should this be `getFirst`?

File: cdap-ui/app/cdap/components/LogViewer/DataFetcher/__tests__/ProgramDataFetcher.test.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+      dataFetcher.getLast().subscribe();
+      expect(prevLogsSpy).toHaveBeenCalledWith(BASE_LOGS_PARAMS);
+    });
+    prevLogsSpy.mockRestore();
+  });
+
+  it('getFirst should fetch logs without offset', () => {
+    const nextLogsSpy = jest.spyOn(MyProgramApi, 'nextLogs');
+    dataFetcher.init().subscribe(() => {
+      dataFetcher.getLast().subscribe();
```
Comment (njbriggs) [inline cdap-ui/app/cdap/components/LogViewer/DataFetcher/__tests__/ProgramDataFetcher.test.ts]: Here too?

File: cdap-ui/app/cdap/components/LogViewer/DataFetcher/__tests__/PreviewDataFetcher.test.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        fromOffset: '2',
+      });
+    });
+    nextLogsSpy.mockRestore();
+  });
+
+  it('should fetch the previous set of logs', () => {
+    const prevLogsSpy = jest.spyOn(MyPreviewApi, 'prevLogs');
+    dataFetcher.init().subscribe(() => {
+      dataFetcher.getNext().subscribe();
```
Comment (njbriggs) [inline cdap-ui/app/cdap/components/LogViewer/DataFetcher/__tests__/PreviewDataFetcher.test.ts]: Should this be `getPrev`? (sorry I missed this before)

Comment (elfenheart) [reply cdap-ui/app/cdap/components/LogViewer/DataFetcher/__tests__/PreviewDataFetcher.test.ts]: fixed for both

**Review by ajainarayanan — APPROVED** [review]

**Pull request merged**